server/proxy-server.js

```diff
@@ -58,31 +58,36 @@ function proxyGet (req, res) {
         .json({'error': 'Target URL is not in the whitelist'});
 
     } else {
-
-      request(url, function (proxyErr, proxyRes) {
-        // Note: proxyRes.body is unparsed data.
-
-        let status = (proxyRes && proxyRes.statusCode) || 500
-        let statusMessage = (proxyRes && proxyRes.statusMessage) || 'No data'
-        let data = (proxyRes && proxyRes.body) || `ERROR: ${statusMessage}`
-
-        if (proxyErr) {
-          status = 500
-          data = proxyErr.toString()
+      
+      // TODO: test error handling
+
+      superagent.get(url)
+      .then(proxyRes => {        
+        console.log(proxyRes.text)
+        
+        function isAValidObject (obj) {
+          return obj && Object.keys(obj).length > 0
         }
+        
+        let status = (proxyRes && proxyRes.statusCode) || 500
```

Comment:
Is this status code override needed? 

Reading https://visionmedia.github.io/superagent/#error-handling, it looks like superagent will only run this block on successful responses and not erroring ones and the final catch will handle error responses from the upstream target). 

Is this for the responses like network failure / timeouts? 
> Network failures, timeouts, and other errors that produce no response will contain no err.status or err.response fields.

Reply:
Nope, the override isn't needed - the `|| 500` should never be triggered under normal circumstances, but I always favour setting a default value just in case.